lib/chef/provider/windows_task.rb

```diff
@@ -194,26 +196,48 @@ def run_schtasks(task_action, options = {})
 
       def task_need_update?
         return true if (new_resource.command &&
-            @current_resource.command != new_resource.command.tr("'", '"')) ||
-            @current_resource.user != new_resource.user ||
-            @current_resource.run_level != new_resource.run_level ||
-            @current_resource.cwd != new_resource.cwd ||
-            @current_resource.frequency_modifier != new_resource.frequency_modifier ||
-            @current_resource.frequency != new_resource.frequency ||
-            @current_resource.idle_time != new_resource.idle_time ||
-            @current_resource.random_delay != new_resource.random_delay ||
-            @current_resource.execution_time_limit != new_resource.execution_time_limit ||
-            !new_resource.start_day.nil? || !new_resource.start_time.nil?
+            current_resource.command != new_resource.command.tr("'", '"')) ||
+            current_resource.user != new_resource.user ||
+            current_resource.run_level != new_resource.run_level ||
+            current_resource.cwd != new_resource.cwd ||
+            current_resource.frequency_modifier != new_resource.frequency_modifier ||
+            current_resource.frequency != new_resource.frequency ||
+            current_resource.idle_time != new_resource.idle_time ||
+            current_resource.random_delay != new_resource.random_delay ||
+            !new_resource.execution_time_limit.include?(current_resource.execution_time_limit) ||
+            (new_resource.start_day && start_day_updated?) ||
+            (new_resource.start_time && start_time_updated?)
         begin
-          return true if new_resource.day.to_s.casecmp(@current_resource.day.to_s) != 0 ||
-              new_resource.months.to_s.casecmp(@current_resource.months.to_s) != 0
+          return true if new_resource.day.to_s.casecmp(current_resource.day.to_s) != 0 ||
+              new_resource.months.to_s.casecmp(current_resource.months.to_s) != 0
         rescue
           Chef::Log.debug "caught a raise in task_needs_update?"
         end
 
         false
       end
 
+      def start_day_updated?
+        current_day = DateTime.parse(current_resource.start_day)
+        new_day = DateTime.parse(new_resource.start_day)
+        current_day != new_day
+      end
+
+      def start_time_updated?
+        time = DateTime.parse(current_resource.start_time).strftime("%H:%M")
+        time != new_resource.start_time
+      end
+
+      def convert_user_date_to_system_date(date_in_string)
```

Comment:
I am suspicious of the need to create a PowerShell process here?

Reply:
see https://github.com/chef/chef/pull/6312#discussion_r136145254